substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/JfrOldObjectSampleEvents.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023, 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, 2023, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.core.genscavenge;
+
+import com.oracle.svm.core.Uninterruptible;
+import com.oracle.svm.core.jfr.HasJfrSupport;
+import com.oracle.svm.core.jfr.JfrEvent;
+import com.oracle.svm.core.jfr.JfrTicks;
+import com.oracle.svm.core.jfr.SubstrateJVM;
+import org.graalvm.word.UnsignedWord;
+
+import java.lang.ref.WeakReference;
+
+public class JfrOldObjectSampleEvents {
```

Comment:
> The class name confused me a bit. I don't think that this class should have Events in its name because it is only used for the sampling (and not for emitting the event).

Makes sense. I can rename it to `JfrOldObjectSampler`.

> allocatedSize should be of type UnsignedWord (if a conversion from UnsignedWord to long is necessary, we try to do it as late as possible).

Hmmmm, this clashes a bit with my attempts to test the profiler logic in a way that it can run in JVM mode (see https://github.com/oracle/graal/pull/6743#discussion_r1335373865). Is it possible to run code that uses `UnsignedWord` in JVM mode? By doing an early switch to long I am then able to keep the `OldObjectProfiler` logic runnable in both JVM and native modes.